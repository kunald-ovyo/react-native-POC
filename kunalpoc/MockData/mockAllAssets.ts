export const mockAllAssetREsponse = [
  {
    name: [
      {
        lang: 'en',
        n: 'carousel',
      },
      {
        lang: 'ta',
        n: 'சரத் ​​குமார்',
      },
    ],
    data: [
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
    ],
  },
  {
    name: [
      {
        lang: 'en',
        n: 'carousel',
      },
      {
        lang: 'ta',
        n: 'சரத் ​​குமார்',
      },
    ],
    data: [
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
    ],
  },
  {
    name: [
      {
        lang: 'en',
        n: 'carousel',
      },
      {
        lang: 'ta',
        n: 'சரத் ​​குமார்',
      },
    ],
    data: [
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
    ],
  },
  {
    name: [
      {
        lang: 'en',
        n: 'carousel',
      },
      {
        lang: 'ta',
        n: 'சரத் ​​குமார்',
      },
    ],
    data: [
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
    ],
  },
  {
    name: [
      {
        lang: 'en',
        n: 'carousel',
      },
      {
        lang: 'ta',
        n: 'சரத் ​​குமார்',
      },
    ],
    data: [
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
    ],
  },
  {
    name: [
      {
        lang: 'en',
        n: 'carousel',
      },
      {
        lang: 'ta',
        n: 'சரத் ​​குமார்',
      },
    ],
    data: [
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
      {
        contentType: 'movie',
        id: '6F40B3A6-FACD-4D2A-AC86-A123CC569A01',
        genres: 'horrer',
        title: [
          {
            lang: 'en',
            n: 'carousel',
          },
          {
            lang: 'ta',
            n: 'சரத் ​​குமார்',
          },
        ],
        rating: '+UA',
      },
    ],
  },
];
