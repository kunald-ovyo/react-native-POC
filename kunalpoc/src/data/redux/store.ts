import {configureStore} from '@reduxjs/toolkit';
//import thunk from 'redux-thunk';

import rootReducer from './RootReducer';

const store = configureStore({
  reducer: rootReducer,
});

export default store;
