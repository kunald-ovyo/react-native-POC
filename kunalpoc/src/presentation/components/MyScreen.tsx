import {View, Text, Button} from 'react-native';
import React, {useEffect, useState} from 'react';

export default function MyScreen() {
  const [state, setState] = useState('Checking');
  useEffect(() => {
    setTimeout(() => {
      console.log('Kunal checking logs ');
    }, 1000);
  }, []);

  return (
    <View>
      <Text testID="myText">{state}</Text>
      <Button
        testID="myButton"
        onPress={() => {
          console.log('Working the logs ');
          setState('changed');
        }}
        title={'Press me'}
      />
    </View>
  );
}
