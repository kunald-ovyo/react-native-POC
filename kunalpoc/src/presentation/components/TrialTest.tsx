import {View, Text} from 'react-native';
import React from 'react';

export default function TrialTest() {
  return (
    <View>
      <Text>TrialTest woenbr</Text>
    </View>
  );
}
