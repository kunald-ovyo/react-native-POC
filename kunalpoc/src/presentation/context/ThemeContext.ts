import {createContext} from 'react';

export interface ThemeAndText {
  primaryColor: string;
  secondaryColor: string;
  backGroundColor: string;
  headingText: {};
}

export const themeContext = createContext<Partial<ThemeAndText> | undefined>(
  undefined,
);

export const ThemeProvider = themeContext.Provider;
