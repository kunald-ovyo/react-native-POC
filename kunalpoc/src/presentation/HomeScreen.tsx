import {StyleSheet} from 'react-native';
import React, {useContext} from 'react';
import {SafeAreaView} from 'react-native-safe-area-context';
import {themeContext} from './context/ThemeContext';
import {ScrollView} from 'react-native-gesture-handler';
import Carousel from './components/Carousel';
import Banner from './components/Banner';
import UsehomeScreenCase from '../domain/customhooks/homeScreenUseCase';

const HomeScreen = () => {
  const themeFontStyleContext = useContext(themeContext);
  const [homeScreenData] = UsehomeScreenCase();

  return (
    <SafeAreaView style={themeFontStyleContext.baseContainerStyle}>
      <ScrollView style={styles.baseScroll}>
        <Carousel data={homeScreenData.carouselData} onClickTile={() => {}} />
        <Banner data={homeScreenData.bannerData} title={'Banner videos'} />
        <Banner data={homeScreenData.bannerData} title={'Banner videos'} />
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  baseScroll: {backgroundColor: 'black', flex: 1},
  baseSkeleton: {
    aspectRatio: 14 / 9,
    backgroundColor: 'white',
    width: '90%',
    margin: '5%',
    borderRadius: 10,
    borderWidth: 1,
    borderColor: '#fff',
  },
  baseHOrrizontalScroll: {
    aspectRatio: 12 / 9,
    backgroundColor: 'white',
    width: '90%',
    margin: '5%',
    borderWidth: 1,
    borderColor: '#fff',
  },
});

export default HomeScreen;
