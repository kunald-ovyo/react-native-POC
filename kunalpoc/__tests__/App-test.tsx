import {act, create} from 'react-test-renderer';
import MyScreen from '../src/presentation/components/MyScreen';
import TrialTest from '../src/presentation/components/TrialTest';
import HomeScreen from '../src/presentation/HomeScreen';
import {Provider} from 'react-redux';
import store from '../src/data/redux/Store';
import {ThemeProvider} from '../src/presentation/context/ThemeContext';
import UsehomeScreenCase from '../src/domain/customhooks/HomeScreenUseCase';
import {mockAllAssetREsponse} from '../MockData/mockAllAssets';

jest.mock('../src/domain/customhooks/HomeScreenUseCase');

UsehomeScreenCase.mockReturnValue([
  {
    allAssetsData: mockAllAssetREsponse,
    carouselData: ['One'],
    bannerData: [],
    loading: false,
  },
]);
// eslint-disable-next-line react/react-in-jsx-scope
const tree = create(<TrialTest />);

// eslint-disable-next-line react/react-in-jsx-scope
const another = create(<MyScreen />);
jest.runAllTimers();

test('Diplsu or compare snapshot', () => {
  expect(tree).toMatchSnapshot();
});

test('Button press ', () => {
  expect(another).toMatchSnapshot();
  const button = another.root.findByProps({testID: 'myButton'}).props;
  act(() => button.onPress());
  const text = another.root.findByProps({testID: 'myText'}).props;
  expect(text.children).toEqual('changed');
});

it('renders correctly', async () => {
  jest.runAllTimers();

  const app = create(
    // eslint-disable-next-line react/react-in-jsx-scope
    <ThemeProvider
      value={{
        primaryColor: 'white',
        secondaryColor: 'yellow',
        backGroundColor: 'black',
        baseContainerStyle: {
          flex: 1,
          flexDirection: 'column',
          backgroundColor: '#17171c',
        },
      }}>
      <Provider store={store}>
        <HomeScreen />
      </Provider>
    </ThemeProvider>,
  );

  // const findAllByTestID = instance =>
  //   instance.findAll(el => el.props.testID === 'banner' && el.type === 'View');
  // expect(findAllByTestID(app).length).toEqual(mockAllAssetREsponse.length - 1);

  const banners = app.root.findAllByProps({testID: 'banner'});
  expect(banners.length / 2).toEqual(mockAllAssetREsponse.length - 1);
});
