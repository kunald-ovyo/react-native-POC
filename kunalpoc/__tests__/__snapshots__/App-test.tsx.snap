// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button press  1`] = `
<View>
  <Text
    testID="myText"
  >
    Checking
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="myButton"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Diplsu or compare snapshot 1`] = `
<View>
  <Text>
    TrialTest woenbr
  </Text>
</View>
`;
